micronota/parsers/cmscan.py

```diff
@@ -0,0 +1,413 @@
+            else:
+                raise CmscanFormatError(
+                    "%s '%s' %s %i. %s" %
+                    ("Unknown strand character",
+                     hitOrientation, "in column",
+                     _COLUMNS['STRAND']['position'],
+                     "Valid characters are '+' for the forward strand and "
+                     "'-' for the reverse strand."))
+
+#           since Infernal want the user to be aware of differences between
+#           CMsearch and CMscan the information about model and query are at
+#           different columns. Here, we contradict this design and store the
+#           information always at the same key
+            if program is 'CMSEARCH':
+                attributes['SEQUENCE_NAME'] = fields[0]
+                attributes['SEQUENCE_ACCESSION'] = fields[1]
+                attributes['MODEL_NAME'] = fields[2]
+                attributes['MODEL_ACCESSION'] = fields[3]
+            elif program is 'CMSCAN':
+                attributes['SEQUENCE_NAME'] = fields[2]
+                attributes['SEQUENCE_ACCESSION'] = fields[3]
+                attributes['MODEL_NAME'] = fields[0]
+                attributes['MODEL_ACCESSION'] = fields[1]
+            else:
+                raise CmscanFormatError("Argument 'program' must be either "
+                                        "'CMSEARCH' or 'CMSCAN'!")
+
+#           iterate through all keys that have not already be handled above
+            for key in _orderedKeys:
+                if key in ['SEQUENCE_START_POSITION',
+                           'SEQUENCE_END_POSITION',
+                           'SEQUENCE_NAME',
+                           'SEQUENCE_ACCESSION',
+                           'MODEL_NAME',
+                           'MODEL_ACCESSION']:
+                    continue
+                attributes[key] = fields[_COLUMNS[key]['position']]
+
+#           cmscan works on multiple sequence in one FASTA file. We want to
+#           yield a separate object for each sequence, thus we create a new
+#           one whenever the ID changes
+            if (currentsequence != attributes['SEQUENCE_NAME'] and
+                    currentsequence is not False):
+                yield annotations
+                annotations = {}
+
+#           a metadata interval is made out of 'frozen dictionary' aka
+#           Features object, where the its hashable value constitutes the
+#           key and the value is a list of tuples, aka intervals
+            annotations[Feature(attributes)] = [(hitStart, hitEnd)]
+#           store current sequence id for the next iteration
+            currentsequence = attributes['SEQUENCE_NAME']
+
+    yield annotations
+
+
+def _writeheader(fh):
+    """ Print a header for cmscan hits. It is similar to 'CMscan --tblout' but
+        not identical, since we do not check for longest occurring names, but
+        use fixed field width. """
```

Comment:
I think this method can be done in one line.  Check out the join command in [string](https://docs.python.org/2/library/string.html)